doc/releases/migration-guide-4.2.rst

```diff
@@ -227,6 +227,15 @@ Stepper
 Bluetooth
 *********
 
+* The buffer types passing through the HCI driver interface are now indicated as H:4 encoded prefix
+  bytes as part of the buffer payload itself. The bt_buf_set_type() and bt_buf_get_type() functions
+  have been deprecated, but are still useable, with the exception that the latter can only be
+  called once per buffer.
+
+* As part of the HCI buffer encoding change, since packet type values are no longer independent
+  bits, the callback used by :c:func:`bt_buf_rx_freed_cb_set` has been changed to take a bitmask
+  that's constructed using bitwise or of ``BIT(type)``` values.
```

Comment:
Should these be under a "Drivers" subsection?

Reply:
> Should these be under a "Drivers" subsection?

I think I'd like to keep it near other Bluetooth item. IIRC we used to have a "Bluetooth HCI" section in past release notes/migration guides, so that might be a good option here.

Reply:
Moved now to a "Bluetooth HCI" section